rnd/autogpt_server/autogpt_server/data/model.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+from typing import Any, ClassVar, Optional
+
+from pydantic import Field, GetCoreSchemaHandler
+from pydantic.fields import FieldInfo
+from pydantic_core import CoreSchema, core_schema
+
+from autogpt_server.util.settings import Secrets
+
+
+class BlockSecret:
+    def __init__(self, key: Optional[str] = None, value: Optional[str] = None):
+        if value is not None:
+            self._value = value
+            return
+
+        self._value = self.__get_secret(key)
+        if self._value is None:
+            raise ValueError(f"Secret {key} not found.")
+
+    STR: ClassVar[str] = "<secret>"
+    SECRETS: ClassVar[Secrets] = Secrets()
+
+    def __repr__(self):
+        return BlockSecret.STR
+
+    def __str__(self):
+        return BlockSecret.STR
+
+    @staticmethod
+    def __get_secret(key: str | None):
+        if not key or not hasattr(BlockSecret.SECRETS, key):
+            return None
+        return getattr(BlockSecret.SECRETS, key)
```

Comment:
Not a fan of this global state.
* it's possible that users may want to use a block multiple times in a flow but with different credentials, e.g. to connect to multiple e-mail accounts
* application-wide state like this isn't compatible with our plans for multi-tenancy

Reply:
Same, I have ideas but this is out of scope here, I just made minimum necessary changes to make tooltips work.